Bugfix: Fix GNU 8.5.0 with OpenMP 4.5 compilation errors

## Describe the Problem ##
I ran into compilation errors when compiling MET version 12.2.1 on a RHEL8 system using GNU version 8.5.0 from 2021. The compilation errors are due to the handle of constant variables in OpenMP clauses. This issue is to patch both the `main_v12.2` and `develop` branches to support compiling MET using earlier versions of GNU.

### Expected Behavior ###
GNU version 8.5.0 from 2021 is recent enough that we should reasonably expect MET to be compiled with it.

### To Reproduce ###
The following code snippet demonstrates the GNU 8.5.0 compilation error:
```
void parse_grid_weight(const Grid &grid, const GridWeightType t,
                       DataPlane &wgt_dp) {
...
#pragma omp parallel default(none) \
   shared(grid, wgt_dp, t)
   {

   // Compute weight for each grid point
#pragma omp for schedule(static) \
                collapse(2)
      for(int x=0; x<grid.nx(); x++) {
         for(int y=0; y<grid.ny(); y++) {

            double w;

            if(t == GridWeightType::Cos_Lat) {
```
Produces this error.
```
apply_mask.cc:128:27: error: ‘t’ is predetermined ‘shared’ for ‘shared’
    shared(grid, wgt_dp, t)
```

Since `t` is a constant argument to the function, GNU 8.5.0 errors out when it's included in the list of shared OpenMP variables:
```
shared(grid, wgt_dp, t)
```
Removing it from that list does satisfy GNU 8.5.0, but then GNU 12.2.0 errors out with:
```
apply_mask.cc: In function ‘void parse_grid_weight(const Grid&, GridWeightType, DataPlane&)’:
apply_mask.cc:139:18: error: ‘t’ not specified in enclosing ‘parallel’
  139 |             if(t == GridWeightType::Cos_Lat) {
      |                ~~^~~~~~~~~~~~~~~~~~~~~~~~~~
apply_mask.cc:127:9: note: enclosing ‘parallel’
  127 | #pragma omp parallel default(none) \
```

The fix is switching from `default(none)` to `default(shared)` and that satisfies both older and new versions of GNU and OpenMP.

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
METplus CCS Project

## Define the Metadata ##

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Review default **alert** labels
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Milestone and Projects ###
- [x] Select **Milestone** as the next bugfix version
- [x] Select **METplus-X.Y Support** project for support of the current coordinated release
- [x] Select **MET-X.Y Development** project for development toward the next coordinated release

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Bugfix Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **main_\<Version>**.
Branch name: `bugfix_<Issue Number>_main_<Version>_<Description>`
- [ ] Fix the bug and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **main_\<Version>**.
Pull request: `bugfix <Issue Number> main_<Version> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the next bugfix version
Select: **METplus-X.Y Support** project for support of the current coordinated release
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Complete the steps above to fix the bug on the **develop** branch.
Branch name:  `bugfix_<Issue Number>_develop_<Description>`
Pull request: `bugfix <Issue Number> develop <Description>`
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the next official version
Select: **MET-X.Y Development** project for development toward the next coordinated release
- [ ] Close this issue.

--- a/src/basic/vx_util/data_cube.cc
+++ b/src/basic/vx_util/data_cube.cc
@@ -198,7 +198,7 @@ void DataCube::add_assign(const DataCube& other) {
    this->check_shape_equal(other);
 
 #pragma omp parallel default(none) \
-   shared(Data, Nxyz, other, bad_data_double)
+   shared(Data, Nxyz, other)
    {
 
 #pragma omp for schedule(static)
@@ -222,7 +222,7 @@ void DataCube::subtract_assign(const DataCube& other) {
    this->check_shape_equal(other);
 
 #pragma omp parallel default(none) \
-   shared(Data, Nxyz, other, bad_data_double)
+   shared(Data, Nxyz, other)
    {
 
 #pragma omp for schedule(static)
@@ -246,7 +246,7 @@ void DataCube::multiply_assign(const DataCube& other) {
    this->check_shape_equal(other);
 
 #pragma omp parallel default(none) \
-   shared(Data, Nxyz, other, bad_data_double)
+   shared(Data, Nxyz, other)
    {
 
 #pragma omp for schedule(static)
@@ -268,7 +268,7 @@ void DataCube::multiply_assign(const DataCube& other) {
 void DataCube::divide_assign(int denom) {
 
 #pragma omp parallel default(none) \
-   shared(Data, Nxyz, denom, bad_data_double)
+   shared(Data, Nxyz, denom)
    {
 
 #pragma omp for schedule(static)
@@ -292,7 +292,7 @@ void DataCube::divide_assign(const DataCube& other) {
    this->check_shape_equal(other);
 
 #pragma omp parallel default(none) \
-   shared(Data, Nxyz, other, bad_data_double)
+   shared(Data, Nxyz, other)
    {
 
 #pragma omp for schedule(static)
@@ -317,7 +317,7 @@ void DataCube::min_assign(const DataCube& other) {
    this->check_shape_equal(other);
 
 #pragma omp parallel default(none) \
-   shared(Data, Nxyz, other, bad_data_double)
+   shared(Data, Nxyz, other)
    {
 
 #pragma omp for schedule(static)
@@ -341,7 +341,7 @@ void DataCube::max_assign(const DataCube& other) {
    this->check_shape_equal(other);
 
 #pragma omp parallel default(none) \
-   shared(Data, Nxyz, other, bad_data_double)
+   shared(Data, Nxyz, other)
    {
 
 #pragma omp for schedule(static)

--- a/src/basic/vx_util/data_plane.cc
+++ b/src/basic/vx_util/data_plane.cc
@@ -466,7 +466,7 @@ int DataPlane::n_good_data() const {
    int n = 0;
 
 #pragma omp parallel default(none) \
-   shared(Data, n)
+   shared(n)
    {
 
       // Count number of good data values
@@ -653,7 +653,7 @@ void DataPlane::standard_anomaly(const DataPlane &mn,
 void DataPlane::replace_bad_data(const double value) {
 
 #pragma omp parallel default(none) \
-   shared(Data, Nxy, value)
+   shared(Data, Nxy)
    {
 
 #pragma omp for schedule(static)
@@ -757,7 +757,7 @@ void DataPlane::data_range(double & data_min, double & data_max) const {
    data_max = -1.0e30;
 
 #pragma omp parallel default(none) \
-   shared(Data, Nxy, data_min, data_max)
+   shared(data_min, data_max)
    {
 
 #pragma omp for schedule(static) \
@@ -791,7 +791,7 @@ MaskPlane DataPlane::mask_plane() const {
    mp.set_size(Nx, Ny);
 
 #pragma omp parallel default(none) \
-   shared(Data, Nxy, mp)
+   shared(mp)
    {
 
 #pragma omp for schedule(static)
@@ -965,7 +965,7 @@ bool DataPlane::fitwav_1d(const int start_wave, const int end_wave) {
    vector<double> S(Nx);
 
 #pragma omp parallel default(none) \
-shared(Nx, Ny, mnw, start_wave, end_wave, C, S)
+shared(Nx, Ny, C, S)
    {
 
 #pragma omp for schedule(static)
@@ -1096,7 +1096,7 @@ DataPlaneArray & DataPlaneArray::operator+=(const DataPlaneArray &d) {
 DataPlaneArray & DataPlaneArray::operator/=(const double v) {
 
 #pragma omp parallel default(none) \
-   shared(Nplanes, Plane, v)
+   shared(Nplanes, Plane)
    {
 
 #pragma omp for schedule(static)

--- a/src/basic/vx_util/data_plane.cc
+++ b/src/basic/vx_util/data_plane.cc
@@ -652,7 +652,7 @@ void DataPlane::standard_anomaly(const DataPlane &mn,
 
 void DataPlane::replace_bad_data(const double value) {
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(Data, Nxy)
    {
 
@@ -964,7 +964,7 @@ bool DataPlane::fitwav_1d(const int start_wave, const int end_wave) {
    vector<double> C(Nx);
    vector<double> S(Nx);
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
 shared(Nx, Ny, C, S)
    {
 
@@ -1095,7 +1095,7 @@ DataPlaneArray & DataPlaneArray::operator+=(const DataPlaneArray &d) {
 
 DataPlaneArray & DataPlaneArray::operator/=(const double v) {
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(Nplanes, Plane)
    {
 

--- a/src/basic/vx_util/data_plane_util.cc
+++ b/src/basic/vx_util/data_plane_util.cc
@@ -114,7 +114,7 @@ void smooth_field(const DataPlane &dp, DataPlane &smooth_dp,
    // For nearest neighbor, no work to do.
    if(width == 1 && mthd == InterpMthd::Nearest) return;
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(mlog, Error, dp, smooth_dp) \
    shared(mthd, width, wrap_lon, t, gaussian) \
    private(v)
@@ -660,7 +660,7 @@ DataPlane normal_cdf_inv(const double area, const DataPlane &mn,
       exit(1);
    }
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(mn, sd, cdf_inv)
    {
 

--- a/src/basic/vx_util/data_plane_util.cc
+++ b/src/basic/vx_util/data_plane_util.cc
@@ -116,7 +116,7 @@ void smooth_field(const DataPlane &dp, DataPlane &smooth_dp,
 
 #pragma omp parallel default(none) \
    shared(mlog, Error, dp, smooth_dp) \
-   shared(mthd, width, shape, wrap_lon, t, gaussian) \
+   shared(mthd, width, wrap_lon, t, gaussian) \
    private(v)
    {
 
@@ -450,7 +450,7 @@ void apply_mask(DataPlane &in, const MaskPlane &mask) {
    int Nxy = mask.nx() * mask.ny();
 
 #pragma omp parallel default(none) \
-   shared(in, mask, Nxy, bad_data_double)
+   shared(in, mask, Nxy)
    {
 
 #pragma omp for schedule(static)
@@ -575,7 +575,7 @@ DataPlane subtract(const DataPlane &dp1, const DataPlane &dp2) {
    }
 
 #pragma omp parallel default(none) \
-   shared(dp1, dp2, diff, bad_data_double)
+   shared(dp1, dp2, diff)
    {
 
 #pragma omp for schedule(static) \
@@ -612,7 +612,7 @@ DataPlane normal_cdf(const DataPlane &dp, const DataPlane &mn,
    }
 
 #pragma omp parallel default(none) \
-   shared(dp, mn, sd, cdf, bad_data_double)
+   shared(dp, mn, sd, cdf)
    {
 
    // Compute the normal CDF for each grid point
@@ -661,7 +661,7 @@ DataPlane normal_cdf_inv(const double area, const DataPlane &mn,
    }
 
 #pragma omp parallel default(none) \
-   shared(area, mn, sd, cdf_inv, bad_data_double)
+   shared(mn, sd, cdf_inv)
    {
 
    // Compute the inverse of the normal CDF for each grid point
@@ -700,7 +700,7 @@ DataPlane gradient(const DataPlane &dp, int dim, int delta) {
    grad_dp.set_constant(bad_data_double);
 
 #pragma omp parallel default(none) \
-   shared(dp, dim, delta, grad_dp, bad_data_double)
+   shared(dp, dim, delta, grad_dp)
    {
 
       // Compute the gradient for each grid point

--- a/src/libcode/vx_data2d/data2d_utils.cc
+++ b/src/libcode/vx_data2d/data2d_utils.cc
@@ -98,7 +98,7 @@ bool derive_wdir(const DataPlane &u2d, const DataPlane &v2d,
    wdir2d.set_constant(bad_data_double);
 
 #pragma omp parallel default(none) \
-   shared(nx, ny, u2d, v2d, wdir2d)
+   shared(u2d, v2d, wdir2d)
    {
 
       //
@@ -164,7 +164,7 @@ bool derive_wind(const DataPlane &u2d, const DataPlane &v2d,
    wind2d.set_constant(bad_data_double);
 
 #pragma omp parallel default(none) \
-   shared(nx, ny, u2d, v2d, wind2d)
+   shared(u2d, v2d, wind2d)
    {
 
       //
@@ -219,7 +219,7 @@ void rotate_wdir_grid_to_earth(const DataPlane &wdir2d, const Grid &g,
    wdir2d_rot.set_constant(bad_data_double);
 
 #pragma omp parallel default(none) \
-   shared(nx, ny, wdir2d, g, wdir2d_rot)
+   shared(wdir2d, g, wdir2d_rot)
    {
 
       //
@@ -292,7 +292,7 @@ bool rotate_uv_grid_to_earth(const DataPlane &u2d, const DataPlane &v2d,
    v2d_rot.set_constant(bad_data_double);
 
 #pragma omp parallel default(none) \
-   shared(nx, ny, u2d, v2d, g, u2d_rot, v2d_rot)
+   shared(u2d, v2d, g, u2d_rot, v2d_rot)
    {
 
       //

--- a/src/libcode/vx_data2d/data2d_utils.cc
+++ b/src/libcode/vx_data2d/data2d_utils.cc
@@ -97,7 +97,7 @@ bool derive_wdir(const DataPlane &u2d, const DataPlane &v2d,
    wdir2d = u2d;
    wdir2d.set_constant(bad_data_double);
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(u2d, v2d, wdir2d)
    {
 
@@ -163,7 +163,7 @@ bool derive_wind(const DataPlane &u2d, const DataPlane &v2d,
    wind2d = u2d;
    wind2d.set_constant(bad_data_double);
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(u2d, v2d, wind2d)
    {
 
@@ -218,7 +218,7 @@ void rotate_wdir_grid_to_earth(const DataPlane &wdir2d, const Grid &g,
    wdir2d_rot = wdir2d;
    wdir2d_rot.set_constant(bad_data_double);
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(wdir2d, g, wdir2d_rot)
    {
 
@@ -291,7 +291,7 @@ bool rotate_uv_grid_to_earth(const DataPlane &u2d, const DataPlane &v2d,
    u2d_rot.set_constant(bad_data_double);
    v2d_rot.set_constant(bad_data_double);
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(u2d, v2d, g, u2d_rot, v2d_rot)
    {
 

--- a/src/libcode/vx_regrid/vx_regrid.cc
+++ b/src/libcode/vx_regrid/vx_regrid.cc
@@ -170,8 +170,7 @@ DataPlane met_regrid_area_weighted(const DataPlane & from_data,
 
 #pragma omp parallel default(none) \
    shared(from_data, from_grid, to_grid, info, to_data) \
-   shared(to_data_sum, wt_data_sum) \
-   shared(bad_data_double)
+   shared(to_data_sum, wt_data_sum)
    { 
 
 #pragma omp single

--- a/src/libcode/vx_regrid/vx_regrid_budget.cc
+++ b/src/libcode/vx_regrid/vx_regrid_budget.cc
@@ -33,7 +33,6 @@ DataPlane met_regrid_budget(const DataPlane & from_data,
 
 #pragma omp parallel default(none) \
    shared(from_data, from_grid, to_grid, info, to_data) \
-   shared(Radius, N, NN, delta) \
    private(count, value, sum)
    {
 

--- a/src/libcode/vx_shapedata/shapedata.cc
+++ b/src/libcode/vx_shapedata/shapedata.cc
@@ -263,7 +263,7 @@ double ShapeData::area_thresh(const ShapeData *raw_ptr,
    const int nxy = data.nxy();
 
    // Number of points inside the object that meet the threshold criteria
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(raw_ptr, obj_thresh, cur_area)
    {
 
@@ -1332,7 +1332,7 @@ void ShapeData::threshold(SingleThresh t) {
    //
    const int nxy = data.nxy();
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(t, data)
    {
 
@@ -1354,7 +1354,7 @@ void ShapeData::set_to_1_or_0() {
 
    const int nxy = data.nxy();
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(data)
    {
 
@@ -1421,7 +1421,7 @@ void ShapeData::threshold_attr(const map<ConcatString,ThreshArray> &attr_map,
    // Zero out discarded shapes
    const int nxy = data.nxy();
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(keep_object, sd_split, data)
    {
 
@@ -1459,7 +1459,7 @@ void ShapeData::threshold_area(SingleThresh t) {
    //
    const int nxy = data.nxy();
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(t, area_object, sd_split, data)
    {
 
@@ -1543,7 +1543,7 @@ void ShapeData::threshold_intensity(const ShapeData *sd_ptr, int perc,
    //
    const int nxy = data.nxy();
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(t, inten_object, sd_split, data)
    {
 
@@ -1714,7 +1714,7 @@ ShapeData select(const ShapeData &id, int n) {
    ShapeData d(id);
    const int nxy = id.data.nxy();
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(id, d, n)
    {
 
@@ -1737,7 +1737,7 @@ void ShapeData::filter(SingleThresh t) {
 
    const int nxy = data.nxy();
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(t, data)
    {
 

--- a/src/libcode/vx_shapedata/shapedata.cc
+++ b/src/libcode/vx_shapedata/shapedata.cc
@@ -264,7 +264,7 @@ double ShapeData::area_thresh(const ShapeData *raw_ptr,
 
    // Number of points inside the object that meet the threshold criteria
 #pragma omp parallel default(none) \
-   shared(raw_ptr, obj_thresh, nxy, data, cur_area)
+   shared(raw_ptr, obj_thresh, cur_area)
    {
 
 #pragma omp for schedule(static) \
@@ -1293,7 +1293,7 @@ void apply_mask(ShapeData &f, const ShapeData &mask) {
    }
 
 #pragma omp parallel default(none) \
-   shared(f, mask, bad_data_float)
+   shared(f, mask)
    {
 
       //
@@ -1333,7 +1333,7 @@ void ShapeData::threshold(SingleThresh t) {
    const int nxy = data.nxy();
 
 #pragma omp parallel default(none) \
-   shared(t, data, nxy)
+   shared(t, data)
    {
 
 #pragma omp for schedule(static)
@@ -1355,7 +1355,7 @@ void ShapeData::set_to_1_or_0() {
    const int nxy = data.nxy();
 
 #pragma omp parallel default(none) \
-   shared(data, nxy)
+   shared(data)
    {
 
 #pragma omp for schedule(static)
@@ -1422,7 +1422,7 @@ void ShapeData::threshold_attr(const map<ConcatString,ThreshArray> &attr_map,
    const int nxy = data.nxy();
 
 #pragma omp parallel default(none) \
-   shared(keep_object, sd_split, data, nxy)
+   shared(keep_object, sd_split, data)
    {
 
 #pragma omp for schedule(static)
@@ -1460,7 +1460,7 @@ void ShapeData::threshold_area(SingleThresh t) {
    const int nxy = data.nxy();
 
 #pragma omp parallel default(none) \
-   shared(t, area_object, sd_split, data, nxy)
+   shared(t, area_object, sd_split, data)
    {
 
 #pragma omp for schedule(static)
@@ -1544,7 +1544,7 @@ void ShapeData::threshold_intensity(const ShapeData *sd_ptr, int perc,
    const int nxy = data.nxy();
 
 #pragma omp parallel default(none) \
-   shared(t, inten_object, sd_split, data, nxy)
+   shared(t, inten_object, sd_split, data)
    {
 
 #pragma omp for schedule(static)
@@ -1715,7 +1715,7 @@ ShapeData select(const ShapeData &id, int n) {
    const int nxy = id.data.nxy();
 
 #pragma omp parallel default(none) \
-   shared(id, d, n, nxy)
+   shared(id, d, n)
    {
 
 #pragma omp for schedule(static)
@@ -1738,7 +1738,7 @@ void ShapeData::filter(SingleThresh t) {
    const int nxy = data.nxy();
 
 #pragma omp parallel default(none) \
-   shared(t, data, nxy)
+   shared(t, data)
    {
 
 #pragma omp for schedule(static)

--- a/src/libcode/vx_statistics/apply_mask.cc
+++ b/src/libcode/vx_statistics/apply_mask.cc
@@ -124,7 +124,7 @@ void parse_grid_weight(const Grid &grid, const GridWeightType t,
    wgt_dp.set_size(grid.nx(), grid.ny());
 
 #pragma omp parallel default(none) \
-   shared(grid, t, wgt_dp)
+   shared(grid, wgt_dp)
    {
 
    // Compute weight for each grid point

--- a/src/libcode/vx_statistics/apply_mask.cc
+++ b/src/libcode/vx_statistics/apply_mask.cc
@@ -123,7 +123,7 @@ void parse_grid_weight(const Grid &grid, const GridWeightType t,
    wgt_dp.clear();
    wgt_dp.set_size(grid.nx(), grid.ny());
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(grid, wgt_dp)
    {
 

--- a/src/libcode/vx_statistics/pair_data_point.cc
+++ b/src/libcode/vx_statistics/pair_data_point.cc
@@ -904,7 +904,7 @@ void apply_mpr_thresh_mask(DataPlane &fcst_dp, DataPlane &obs_dp,
    bool ocmn_flag = !(ocmn_dp.is_empty());
    bool ocsd_flag = !(ocsd_dp.is_empty());
 
-#pragma omp parallel default(none)                                \
+#pragma omp parallel default(shared)                              \
    shared(fcst_dp, obs_dp, fcmn_dp, fcsd_dp, ocmn_dp, ocsd_dp, m) \
    shared(nxy, n_skip, fcmn_flag, fcsd_flag, ocmn_flag, ocsd_flag)
    {

--- a/src/libcode/vx_statistics/pair_data_point.cc
+++ b/src/libcode/vx_statistics/pair_data_point.cc
@@ -904,10 +904,9 @@ void apply_mpr_thresh_mask(DataPlane &fcst_dp, DataPlane &obs_dp,
    bool ocmn_flag = !(ocmn_dp.is_empty());
    bool ocsd_flag = !(ocsd_dp.is_empty());
 
-#pragma omp parallel default(none)                                 \
-   shared(fcst_dp, obs_dp, fcmn_dp, fcsd_dp, ocmn_dp, ocsd_dp, m)  \
-   shared(nxy, n_skip, fcmn_flag, fcsd_flag, ocmn_flag, ocsd_flag) \
-   shared(bad_data_double)
+#pragma omp parallel default(none)                                \
+   shared(fcst_dp, obs_dp, fcmn_dp, fcsd_dp, ocmn_dp, ocsd_dp, m) \
+   shared(nxy, n_skip, fcmn_flag, fcsd_flag, ocmn_flag, ocsd_flag)
    {
 
       // Loop over the pairs

--- a/src/tools/core/grid_stat/grid_stat.cc
+++ b/src/tools/core/grid_stat/grid_stat.cc
@@ -2978,7 +2978,7 @@ void write_nc(const ConcatString &field_name, const DataPlane &dp,
          &conf_info.mask_map[conf_info.vx_opt[i_vx].mask_name[i]];
 
 #pragma omp parallel default(none) \
-      shared(grid, DefaultTO, apply_mask, mask_ptr, data, dp, bad_data_float)
+      shared(grid, DefaultTO, apply_mask, mask_ptr, data, dp)
       {
 
       // Store the data
@@ -3140,7 +3140,7 @@ void write_nbrhd_nc(const DataPlane &fcst_dp, const DataPlane &obs_dp,
 
 #pragma omp parallel default(none) \
    shared(grid, DefaultTO, apply_mask, mask_ptr) \
-   shared(fcst_data, obs_data, fcst_dp, obs_dp, bad_data_float)
+   shared(fcst_data, obs_data, fcst_dp, obs_dp)
    {
 
       // Store the forecast and observation values

--- a/src/tools/core/mode/combine_boolplanes.cc
+++ b/src/tools/core/mode/combine_boolplanes.cc
@@ -60,7 +60,7 @@ void boolplane_to_pgm(const BoolPlane & in, Pgm & out) {
    out.all_white();
 
 #pragma omp parallel default(none) \
-   shared(out, in, black, white)
+   shared(out, in)
    {
 
 #pragma omp for schedule(static) \

--- a/src/tools/core/mode/combine_boolplanes.cc
+++ b/src/tools/core/mode/combine_boolplanes.cc
@@ -59,7 +59,7 @@ void boolplane_to_pgm(const BoolPlane & in, Pgm & out) {
    out.set_size_xy(in.nx(), in.ny());
    out.all_white();
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(out, in)
    {
 

--- a/src/tools/core/mode/mode_exec.cc
+++ b/src/tools/core/mode/mode_exec.cc
@@ -1961,7 +1961,7 @@ void ModeExecutive::write_obj_netcdf(const ModeNcOutInfo & info)
    }
 
 #pragma omp parallel default(none) \
-   shared(grid, info, DefaultTO, unmatched_id, bad_data_int) \
+   shared(grid, info, DefaultTO) \
    shared(fcst_raw_data, obs_raw_data) \
    shared(fcst_obj_raw_data, fcst_obj_data) \
    shared(obs_obj_raw_data, obs_obj_data) \

--- a/src/tools/core/mode/mode_ps_file.cc
+++ b/src/tools/core/mode/mode_ps_file.cc
@@ -1325,7 +1325,7 @@ Box valid_xy_bb(const ShapeData * wd_ptr, const Grid & grid)
    const int data_nx = wd_ptr->data.nx();
    const int data_ny = wd_ptr->data.ny();
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(wd_ptr) \
    shared(L, B, R, T, N)
    {

--- a/src/tools/core/mode/mode_ps_file.cc
+++ b/src/tools/core/mode/mode_ps_file.cc
@@ -1326,7 +1326,7 @@ Box valid_xy_bb(const ShapeData * wd_ptr, const Grid & grid)
    const int data_ny = wd_ptr->data.ny();
 
 #pragma omp parallel default(none) \
-   shared(data_nx, data_ny, wd_ptr) \
+   shared(wd_ptr) \
    shared(L, B, R, T, N)
    {
 

--- a/src/tools/core/mode/mode_superobject.cc
+++ b/src/tools/core/mode/mode_superobject.cc
@@ -186,7 +186,6 @@ ModeSuperObject::ModeSuperObject(bool isFcst, int n_files, bool do_clusters,
    auto merge_sd = ShapeData(*(mvd[0]->_simple->_sd));
 
 #pragma omp parallel default(none) \
-   shared(nx, ny, bad_data_double) \
    shared(_simple_result, _simple_sd) \
    shared(merge_result, merge_sd)
    {

--- a/src/tools/core/mode/mode_superobject.cc
+++ b/src/tools/core/mode/mode_superobject.cc
@@ -185,7 +185,7 @@ ModeSuperObject::ModeSuperObject(bool isFcst, int n_files, bool do_clusters,
    _simple_sd    = ShapeData(*(mvd[0]->_simple->_sd));
    auto merge_sd = ShapeData(*(mvd[0]->_simple->_sd));
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(_simple_result, _simple_sd) \
    shared(merge_result, merge_sd)
    {

--- a/src/tools/core/mode/multivar_data.cc
+++ b/src/tools/core/mode/multivar_data.cc
@@ -334,7 +334,7 @@ static void populate_bool_plane(const string &name, const int * buf,
    int nyes=0;
 
 #pragma omp parallel default(none) \
-   shared(nx, ny, buf, nyes, bp_out)
+   shared(buf, nyes, bp_out)
    {
 
 #pragma omp for schedule(static) \

--- a/src/tools/core/mode/multivar_data.cc
+++ b/src/tools/core/mode/multivar_data.cc
@@ -333,7 +333,7 @@ static void populate_bool_plane(const string &name, const int * buf,
 
    int nyes=0;
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(buf, nyes, bp_out)
    {
 

--- a/src/tools/core/mode/objects_from_netcdf.cc
+++ b/src/tools/core/mode/objects_from_netcdf.cc
@@ -142,7 +142,7 @@ static void populate_bool_plane(const int * buf, const int nx, const int ny, Boo
    int nyes = 0.0;
 
 #pragma omp parallel default(none) \
-   shared(nx, ny, buf, nyes, bp_out)
+   shared(buf, nyes, bp_out)
    {
 
 #pragma omp for schedule(static) \

--- a/src/tools/core/mode/objects_from_netcdf.cc
+++ b/src/tools/core/mode/objects_from_netcdf.cc
@@ -141,7 +141,7 @@ static void populate_bool_plane(const int * buf, const int nx, const int ny, Boo
 
    int nyes = 0.0;
 
-#pragma omp parallel default(none) \
+#pragma omp parallel default(shared) \
    shared(buf, nyes, bp_out)
    {
 

--- a/src/tools/core/pcp_combine/pcp_combine.cc
+++ b/src/tools/core/pcp_combine/pcp_combine.cc
@@ -1204,7 +1204,7 @@ static void do_derive_command() {
          der_dp = max_dp;
 
 #pragma omp parallel default(none) \
-   shared(nxy, max_dp, min_dp, bad_data_double, der_dp)
+   shared(nxy, max_dp, min_dp, der_dp)
          {
 
 #pragma omp for schedule(static)
@@ -1226,7 +1226,7 @@ static void do_derive_command() {
          der_dp = sum_dp;
 
 #pragma omp parallel default(none) \
-   shared(nxy, sum_dp, vld_dp, bad_data_double, der_dp)
+   shared(nxy, sum_dp, vld_dp, der_dp)
          {
 
 #pragma omp for schedule(static)
@@ -1249,7 +1249,7 @@ static void do_derive_command() {
          der_dp = sum_dp;
 
 #pragma omp parallel default(none) \
-   shared(nxy, sum_dp, sum_sq_dp, vld_dp, bad_data_double, der_dp) \
+   shared(nxy, sum_dp, sum_sq_dp, vld_dp, der_dp) \
    private(v)
          {
 

--- a/src/tools/other/gen_ens_prod/gen_ens_prod.cc
+++ b/src/tools/other/gen_ens_prod/gen_ens_prod.cc
@@ -620,8 +620,7 @@ static void get_ens_mean_stdev(GenEnsProdVarInfo *ens_info,
 #pragma omp parallel default(none) \
    shared(nxy, emn_dp, esd_dp) \
    shared(emn_cnt, emn_sum) \
-   shared(esd_cnt, esd_sum, esd_ssq) \
-   shared(bad_data_double)
+   shared(esd_cnt, esd_sum, esd_ssq)
    {      
 
 #pragma omp for schedule(static)
@@ -861,7 +860,7 @@ static void write_ens_nc(GenEnsProdVarInfo *ens_info, int n_ens_vld,
    double t = conf_info.vld_data_thresh;
 
 #pragma omp parallel default(none) \
-   shared(ens_vld, n_ens_vld, t, bad_data_float) \
+   shared(ens_vld, n_ens_vld, t) \
    shared(ens_mean, ens_stdev, ens_minus, ens_plus, ens_min, ens_max, ens_range) \
    shared(sum_na, cnt_na, min_na, max_na, stdev_sum_na, stdev_ssq_na, stdev_cnt_na)
    {

--- a/src/tools/other/shift_data_plane/shift_data_plane.cc
+++ b/src/tools/other/shift_data_plane/shift_data_plane.cc
@@ -258,7 +258,7 @@ void process_data_file() {
 
 #pragma omp parallel default(none) \
    shared(dp_shift, dp_in, dx, dy, grid) \
-   shared(Method, Width, Shape, bad_data_double)
+   shared(Method, Width, Shape)
    {
 
       // Shift the data

--- a/src/tools/tc_utils/tc_dland/tc_dland.cc
+++ b/src/tools/tc_utils/tc_dland/tc_dland.cc
@@ -256,7 +256,7 @@ void process_distances() {
         << " points in grid (" << grid.serialize() << ")...\n";
 
 #pragma omp parallel default(none) \
-   shared(DefaultTO, grid, dland, land_array, nautical_miles_per_km)
+   shared(DefaultTO, grid, dland, land_array)
    {
 
       // Loop over the grid and compute the distance to land for each point   

